File: Final 2018-3/srcgen/Producto.rb
	
class Producto

	def initialize(this,tipo,material,identificacion)
		@tipo = tipo
		@material = material
		@identificacion = identificacion
	end
	
	def tipo
		return @tipo
	end

	def material
		return @material
	end

	def identificacion
		return @identificacion
	end

    def tipo=(tipo)
		@tipo = tipo
    end
	
    def material=(material)
		@material = material
    end
	
    def identificacion=(identificacion)
		@identificacion = identificacion
    end
	
end

File: Final 2018-3/srcgen/Producto.cpp
class Producto{

  Producto(this){
  }
  Producto((this,tipo,material,identificacion):
      this.tipo=tipo
      this.material=material
      this.identificacion=identificacion

  void getTipo(){
    return this.tipo
	}
  void getMaterial(){
    return this.material
	}
  void getIdentificacion(){
    return this.identificacion
	}
  String setTipo(this,tipo){
    this.tipo = tipo
  }

  String setMaterial(this,material){
    this.material = material
  }

  String setIdentificacion(this,identificacion){
    this.identificacion = identificacion
  }

};